PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -224,10 +226,10 @@ public function read($len)
                 if ($this->canDispatchPcntlSignal) {
                     // prevent cpu from being consumed while waiting
                     if ($this->canSelectNull) {
```

Comment:
Seems `canSelectNull` becomes redundant with this PR and we can just always do`select`.

Also, could you please pay some attention to https://github.com/php-amqplib/php-amqplib/commit/6200ed98499429f2d655ff6cffe63ca6152c1a89? Seems this commit completely breaks the logic of the piece of code we're currently considering. It turns `select (null, null)` into `select (0, 0)` which results in completely different behaviour of `stream_select`(returns immediately).